pipeline/id_minter/src/main/scala/uk/ac/wellcome/platform/idminter/database/IdentifiersDao.scala

```diff
@@ -25,68 +25,73 @@ object IdentifiersDao {
 class IdentifiersDao(identifiers: IdentifiersTable) extends Logging {
   import IdentifiersDao._
 
-  def withConnection[R](doWork: DBSession => R): R =
-    DB localTx doWork
-
   def lookupIds(sourceIdentifiers: Seq[SourceIdentifier])(
-    implicit session: DBSession): Try[LookupResult] =
+    implicit session: DBSession = readOnlySession): Try[LookupResult] =
     Try {
       debug(s"Matching ($sourceIdentifiers)")
+      val distinctIdentifiers = sourceIdentifiers.distinct
       val identifierRowsMap: Map[(String, String, String), SourceIdentifier] =
-        sourceIdentifiers.map { id =>
+        distinctIdentifiers.map { id =>
           (id.ontologyType, id.identifierType.id, id.value) -> id
         }.toMap
 
-      blocking {
-
-        val i = identifiers.i
-        val matchRow = rowMatcherFromSyntaxProvider(i) _
-
-        // The query is manually constructed like this because using WHERE IN
-        // statements with multiple items causes a full-index scan on MySQL 5.6.
-        // See: https://stackoverflow.com/a/53180813
-        // Therefore, we perform the rewrites here on the client.
-        //
-        // Because of issues in `scalikejdbc`'s typing, it's necessary to
-        // specify the `Nothing` type parameters as done here in order for
-        // the `map` statement to work properly.
-        val query = withSQL[Nothing] {
-          select
-            .from[Nothing](identifiers as i)
-            .where
-            .map { query =>
-              sourceIdentifiers.tail.foldLeft(
-                query.withRoundBracket(matchRow(sourceIdentifiers.head))
-              ) { (q, sourceIdentifier) =>
-                q.or.withRoundBracket(matchRow(sourceIdentifier))
+      val foundIdentifiers =
+        withTimeWarning(thresholdMillis = 10000L, distinctIdentifiers) {
+          batchSourceIdentifiers(distinctIdentifiers)
+            .flatMap { identifierBatch =>
+              blocking {
+                val i = identifiers.i
+                val matchRow = rowMatcherFromSyntaxProvider(i) _
+
+                // The query is manually constructed like this because using WHERE IN
+                // statements with multiple items causes a full-index scan on MySQL 5.6.
+                // See: https://stackoverflow.com/a/53180813
+                // Therefore, we perform the rewrites here on the client.
+                //
```

Comment:
Aren't we using Aurora, with mysql compatibility?

Reply:
Good question. The short answer is yes. The long answer after reading about it is yes, but the Aurora versions are actually forks of the corresponding MySQL, Postgres, etc versions where Amazon have replaced the underlying Things What Do Storage but have left the query, indexing, etc stuff completely intact - so the compatibility is up to and including bugs and other issues.